pkgs/desktops/gnome/extensions/nautilus-open-any-terminal/default.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, pkg-config
+, python3Packages
+, dbus
+, dconf
+, gobject-introspection
+, substituteAll
+, fetchFromGitHub
+, gnome3
+, glib
+, gtk3
+, python3
+, gsettings-desktop-schemas
+, wrapGAppsHook
+}:
+
+# nix-env-pwd-install gnomeExtensions.nautilus-open-any-terminal
+
+python3Packages.buildPythonPackage rec {
+  pname = "nautilus-open-any-terminal";
+  version = "0.2.15";
+
+  src = fetchFromGitHub {
+    owner = "Stunkymonkey";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-cc6Lh5XeAuU5Os4eJ0QcL6XJYB6DqxeUGaOf6m1OnpY=";
+  };
+
+  nativeBuildInputs = [
+    glib
+    gobject-introspection
+    pkg-config
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    dbus
+    dconf
+    gnome3.nautilus
+    gnome3.nautilus-python
+    gsettings-desktop-schemas
+    gtk3
+    python3
+    python3.pkgs.pygobject3
+    gobject-introspection
+  ];
+
+  patches = [ ./hardcode-gsettings.patch ];
+
+  postPatch = ''
+    substituteInPlace nautilus_open_any_terminal/open_any_terminal_extension.py \
+      --subst-var-by gsettings_path ${glib.makeSchemaPath "$out" "$name"}
+  '';
+
+  postInstall = ''
+    glib-compile-schemas "$out/share/glib-2.0/schemas"
+  '';
+
+  PKG_CONFIG_LIBNAUTILUS_EXTENSION_EXTENSIONDIR = "${placeholder "out"}/lib/nautilus/extensions-3.0";
+
+  doCheck = false;
+
+  passthru = {
+    extensionUuid = "nautilus-open-any-terminal@stunkymonkey.github.com";
+    extensionPortalSlug = "nautilus-open-any-terminal";
+  };
+
+  meta = with lib; {
+    description = "extension for nautilus, which adds an context-entry for opening other terminal-emulators then `gnome-terminal`.";
+    license = licenses.gpl3;
```

Comment:
`gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.